aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1550,17 +1551,18 @@ inline void baddbmm_cpu_kernel(const Tensor& result, const Tensor& self, const T
           auto r2 = r1[i];
           auto s2 = s1[i];
           for (const auto j : c10::irange(js)) {
-            scalar_t &r = r2[j];
+            opmath_t acc_value = 0;//is_bmm ? opmath_t(0) : opmath_t(r2[j]);
+            for (const auto k : c10::irange(ks)) {
+              acc_value += s2[k] * m1[k][j];
```

Comment:
```suggestion
              acc_value += static_cast<opmath_t>(s2[k]) * static_cast<opmath_t>(m1[k][j]);
```

Reply:
Done.
Just for curiosity, does this change is for readable only? The `bfloat16` type should have overrided `*` so the cast here should happen whether we add `static_cast<opmath_t>` or not.

Reply:
`BFloat16::operator*` casts the result back to `BFloat16`, so this cast increases precision.

Reply:
Got it, thanks.